### ISSUE TITLE
double typedef in toxav/rtp.h

### ISSUE THREAD

**Issue by azarens** [irungentoo/toxcore#1497]
There's a double typedef in rtp.h of `__fail_if_misaligned`.  Specifically,

On line 68:
```
/* Check alignment */
typedef char __fail_if_misaligned [ sizeof(struct RTPHeader) == 80 ? 1 : -1 ];
```

And also on line 78:
```
/* Check alignment */
typedef char __fail_if_misaligned [ sizeof(struct RTPMessage) == 82 ? 1 : -1 ];
```

This results in a compile error with gcc 4.5.3 on my machine.

What is the purpose of this double declaration?  Would a workaround be possible?

**Comment by cebe** [comment]
can you show the error?

**Comment by azarens** [comment]
```
--- ../other/DHT_bootstrap-bootstrap_node_packets.o ---
  CC       ../other/DHT_bootstrap-bootstrap_node_packets.o
--- ../toxav/libtoxav_la-rtp.lo ---
  CC       ../toxav/libtoxav_la-rtp.lo
--- ../toxav/libtoxav_la-msi.lo ---
  CC       ../toxav/libtoxav_la-msi.lo
--- ../toxav/libtoxav_la-group.lo ---
  CC       ../toxav/libtoxav_la-group.lo
--- ../toxav/libtoxav_la-rtp.lo ---
In file included from ../toxav/rtp.c:26:0:
../toxav/rtp.h:78:14: error: redefinition of typedef '__fail_if_misaligned'
../toxav/rtp.h:68:14: note: previous declaration of '__fail_if_misaligned' was here
*** [../toxav/libtoxav_la-rtp.lo] Error code 1

make: stopped in /usr/obj/wip/toxcore/work/toxcore/build
1 error

make: stopped in /usr/obj/wip/toxcore/work/toxcore/build
*** [all-recursive] Error code 1

make: stopped in /usr/obj/wip/toxcore/work/toxcore
1 error

```

**Comment by azarens** [comment]
Just verified that the package builds successfully with gcc 4.9.

cf. https://gcc.gnu.org/wiki/C11Status

Should a configure script be modified to fail in case a sufficient version of gcc is not available?

**Comment by mannol** [comment]
That should cover it.

**Comment by cebe** [comment]
should be fixed by #1499 -> close
